sdk/core/src/services/github/client/mod.rs

```diff
@@ -225,42 +258,113 @@ pub struct Repo {
     /// hash of the default branch
     #[serde(default = "empty_string")]
     pub last_hash: String,
-}
-
-/// Little function to define default booleans
-/// for struct values that are to be used to collect Json
-fn default_bool() -> bool {
-    false
-}
-
-/// Little function to define default Strings
-/// for struct values that are to be used to collect Json
-fn empty_string() -> String {
-    "".to_string()
+    /// Latest release tag name
+    #[serde(default = "empty_string")]
+    pub version: String,
 }
 
 /// Repo impl
 impl Repo {
+    /// Conventional Constructor
+    pub fn new() -> Self {
+        Repo {
+            full_name: None,
+            html_url: None,
+            default_branch: None,
+            language: None,
+            archived: None,
+            disabled: None,
+            empty: default_bool(),
+            last_hash: empty_string(),
+            version: empty_string(),
+        }
+    }
+
     /// This method allows us to add the last hash to a
     /// Repo if it is newer that what is already in Mongo
-    ///
     pub fn add_last_hash(&mut self, last_hash: String) {
         self.last_hash = last_hash;
     }
 
+    /// Method to add a version
+    pub fn add_version(&mut self, version: String) {
+        self.version = version;
+    }
+
     /// Method to mark the Repository as empty
-    ///
     pub fn mark_repo_empty(&mut self) {
         self.empty = true;
     }
 }
 
+/// Default Implementation
+impl Default for Repo {
+    fn default() -> Self {
+        Repo::new()
+    }
+}
+
+/// The Release struct is used to collect data from a response
+/// from the /repos/{org}/{repo_name}/releases/latest endpoint.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Release {
+    /// The tag name associated to the release
+    #[serde(default = "default_version")]
+    pub tag_name: String,
+}
+
+impl Release {
+    fn new() -> Self {
+        Release {
+            tag_name: default_version(),
+        }
+    }
+}
+
+/// Little function to define default booleans
+/// for struct values that are to be used to collect Json
+fn default_bool() -> bool {
+    false
+}
+
+/// Little function to define default Strings
+/// for struct values that are to be used to collect Json
+fn empty_string() -> String {
+    "".to_string()
+}
+
+/// Function used to define a default
```

Comment:
Since this is a special magic string, how do you feel about adding some explanation as to what is means, and how we came up with it?  TBH, I suggested this in conversation, but I'm not sold its the best possible magic string. I'd definitely be open to better suggestions that are more explicit.

Reply:
Yes, that is a great idea.  I thought at first maybe "default" might be better, but it lacks the structure of a semver, so I went back to "d0.0.0".  I think having that structure makes it easier to understand that you are looking at a version which is important to be able to do when looking quickly and the odd 'd' in front of it will beg the reader to ask that it means if they don't already know.  I think that's better than most of the thoughts I've had about what the version should be - looking at the spec